x-pack/plugins/security_solution/common/endpoint/schema/resolver.ts

```diff
@@ -23,6 +23,37 @@ export const validateTree = {
   }),
 };
 
+/**
+ * Used to validate GET requests for a complete resolver tree.
+ */
+export const validateTree = {
+  body: schema.object({
+    // if the ancestry field is specified this field will be ignored
+    descendantLevels: schema.number({ defaultValue: 20, min: 0, max: 1000 }),
```

Comment:
Remind me why the ancestry affects the descendantLevels again?

Reply:
Yeah it affects it because if we have the `ancestry` field defined than we have a much more performant way of retrieving levels so my thought was to not _**limit**_ the number of levels that come back in that scenario. We could still limit it, but what we'd likely have to do is get all the levels back like we normally do with the ancestry array, bucket them together by level, and then remove the levels that exceeded the requested number which seems kind of wasteful.

I'll add this as a comment as well 👍